tests/js/client/shell/transaction/replication2_recovery/shell-transaction-replication2-recovery.js

```diff
@@ -0,0 +1,276 @@
+      let logContents = replicatedLogsHelper.dumpShardLog(shardId);
+      let abortAllEntryFound = _.some(logContents, entry => {
+        if (entry.logTerm !== newTerm || entry.payload === undefined) {
+          return false;
+        }
+        return entry.payload[1].operation === "AbortAllOngoingTrx";
+      });
+      assertTrue(abortAllEntryFound);
+
+      // Expect further transaction operations to fail.
+      try {
+        tc.insert({ _key: 'test3', value: 3 });
+        fail('Insert was expected to fail due to transaction abort.');
+      } catch (ex) {
+        assertEqual(internal.errors.ERROR_TRANSACTION_NOT_FOUND.code, ex.errorNum);
+      }
+
+      syncShardsWithLogs(dbn);
+
+      // Try a new transaction, this time expecting it to work.
+      try {
+        trx = db._createTransaction({
+          collections: { write: c.name() }
+        });
+        tc = trx.collection(c.name());
+        tc.insert({ _key: "foo" });
+        trx.commit();
+      } catch (err) {
+        fail("Transaction failed with: " + JSON.stringify(err));
+      }
+
+      let servers = Object.assign({}, ...followers.map(
+        (serverId) => ({ [serverId]: replicatedLogsHelper.getServerUrl(serverId) })));
+      for (const [key, endpoint] of Object.entries(servers)) {
+        replicatedLogsHelper.waitFor(
+          replicatedStatePredicates.localKeyStatus(endpoint, dbn, shardId, "foo", true));
+      }
+
+      // Resume the dead server. Expect to read "foo" from it.
+      continueServerWait(leader);
+      syncShardsWithLogs(dbn);
+      replicatedLogsHelper.waitFor(
+        replicatedStatePredicates.localKeyStatus(
+          replicatedLogsHelper.getServerUrl(leader), dbn, shardId, "foo", true), 30);
+
+      // Try another transaction. This time expect it to work on all servers.
+      try {
+        trx = db._createTransaction({
+          collections: { write: c.name() }
+        });
+        tc = trx.collection(c.name());
+        tc.insert({ _key: "bar" });
+        trx.commit();
+      } catch (err) {
+        fail("Transaction failed with: " + JSON.stringify(err));
+      }
+
+      servers = Object.assign({}, ...participants.map(
+        (serverId) => ({ [serverId]: replicatedLogsHelper.getServerUrl(serverId) })));
+      for (const [_, endpoint] of Object.entries(servers)) {
+        replicatedLogsHelper.waitFor(
+          replicatedStatePredicates.localKeyStatus(endpoint, dbn, shardId, "bar", true));
+      }
+    },
+
+    /**
+     * This test is disabled because currently we can't properly stop the maintenance from deleting the shard
+     * on the newly added follower. When the core is constructed, the follower tries to create the new shard locally,
+     * but because the server is not listed as a participant for that shard in Plan/Collections, the maintenance figures
+     * the shard shouldn't exist locally.
+     */
+    DISABLED_testFollowerReplaceDuringTransaction: function () {
```

Comment:
In general i like this test, i would also like to have a pretty similar version, where we kill all followers, such that the write concern cannot be reached anymore, and assert that we have as designed in this case.

Reply:
I noted this idea in: https://arangodb.atlassian.net/browse/CINFRA-531